cmd/genrconfig/main.go

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package main
+
+import (
+  "os"
+  "fmt"
+  "log"
+  "io/ioutil"
+  "strings"
+  
+  "github.com/googleinterns/executable-mocks/cmd/bldmock"
+  "github.com/golang/protobuf/proto"
+)
+
+/*
+ * Takes command line arguments and generates a Protocol Buffer configuration file.
+ * An example of use is: ./genrconfig configs/example name "strarg:generic string" "infile:tmp/input"
+ * The first argument defines the path of the configuration file. The extension .textproto is added.
```

Comment:
A typical design for Unix command line is to simply print the result to the standard output.

So instead of getting a file as input, you can just print the result (the generated proto) directly to the standard output. The way one would then use your command is:

    ./genrconfig "strarg:generic string" "infile:tmp/input" > configs/example name

That gives some more flexibility to the user, because they can potentially combine or process the proto directly, instead of having to go through a file.

Also usually I would advise against adding an implicit extension like ".textproto", file content should be independent of filename/extension, and everything should work the same with or without the extension.

Reply:
I think that are great ideas.